[Lens as code] Revisit metric chart schema

dej611 opened the pull request:
## Summary

Fix https://github.com/elastic/kibana/issues/244506

This PR fixes few Metric chart API shortcomings:

* Breakdown dimension can be ranked by secondary metric now
  * due to `schema` library lack of tuple support, while the `metrics` property can be an array, the introduction of the `type` discrimination property had to be done to avoid random default injections.
* Fixed background chart schema for ES|QL version
* Added more E2E tests (both from SO and API versions)


### Checklist

- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

dej611 committed:
:bug: Revisit metric schema to handle background chart and rank by secondary metric too

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/375616)
* Commit: f310f20702063b89e9573c37942c21d165620d46

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/375616#019b3726-be6e-4327-b267-974a9b679853)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375616","number":375616,"commit":"f310f20702063b89e9573c37942c21d165620d46"}],"number":375616}
buildkite-pr-comment-->

dej611 committed:
:label: Fix types

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/375659)
* Commit: 5aa2c9b4963eafd0dc4f3ee256bad5272948a64a

### Failed CI Steps
* [Jest Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b5d5-4405-83d3-b38099be6244)
* [x-pack/platform/test/serverless/functional/configs/observability/config.group8.ts](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b5e2-48c3-9457-3389e65fb764)
* [x-pack/platform/test/serverless/functional/configs/search/config.group8.ts](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b5e3-40c2-8f94-d0ea3767292a)
* [x-pack/platform/test/serverless/functional/configs/security/config.group8.ts](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b5e4-480a-9fb0-a4a1fda80fd9)
* [FTR Configs #5](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b5e9-45a2-bc2f-dc2cacfadfc4)
* [FTR Configs #13](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b5f0-429c-840f-7b0fa783d771)
* [FTR Configs #38](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b609-4b1e-bedc-d6d69ba57389)
* [FTR Configs #50](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b616-4ff8-9b7d-5e34ee8cc585)
* [FTR Configs #131](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3777-b650-4b42-a23c-ab5c558eeb0f)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b3758-0b21-4996-a021-5375b3a45f81)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/375659#019b377a-c1a3-459c-8476-60302fbf4b37)

### History
* :broken_heart: [Build #375616](https://buildkite.com/elastic/kibana-pull-request/builds/375616) failed f310f20702063b89e9573c37942c21d165620d46

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375659","number":375659,"commit":"5aa2c9b4963eafd0dc4f3ee256bad5272948a64a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375616","number":375616,"commit":"f310f20702063b89e9573c37942c21d165620d46"}],"number":375659}
buildkite-pr-comment-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-entity-store-performance-from-pr --> Click to trigger **kibana-entity-store-performance-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/375722)
* Commit: 5aa2c9b4963eafd0dc4f3ee256bad5272948a64a

### Failed CI Steps
* [x-pack/platform/test/serverless/functional/configs/observability/config.group8.ts](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37b5-2e83-4449-94f6-bf4d0b7446e8)
* [x-pack/platform/test/serverless/functional/configs/search/config.group8.ts](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37b5-2e85-4795-a1e6-eb55b228f650)
* [x-pack/platform/test/serverless/functional/configs/search/config.group8.ts](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37bc-26f0-4590-b03f-cb9475ccc198)
* [x-pack/platform/test/serverless/functional/configs/security/config.group8.ts](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37b5-2e85-4332-8651-d9da38920577)
* [x-pack/platform/test/serverless/functional/configs/security/config.group8.ts](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37bc-f5df-40ac-b938-28362ede1c7f)
* [FTR Configs #10](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37b5-2e8f-4b41-a9e5-ee5ea2ac0087)
* [FTR Configs #55](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37b5-2ebf-4cd3-8a16-1d5d71245cff)
* [FTR Configs #60](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37b5-2ec4-45dc-9804-de39ea702509)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b3799-5a7b-407c-9633-6f0280c27988)
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/375722#019b37b6-59d7-47c6-b736-447b4d871830)

### History
* :broken_heart: [Build #375659](https://buildkite.com/elastic/kibana-pull-request/builds/375659) failed 5aa2c9b4963eafd0dc4f3ee256bad5272948a64a
* :broken_heart: [Build #375616](https://buildkite.com/elastic/kibana-pull-request/builds/375616) failed f310f20702063b89e9573c37942c21d165620d46

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375722","number":375722,"commit":"5aa2c9b4963eafd0dc4f3ee256bad5272948a64a"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375659","number":375659,"commit":"5aa2c9b4963eafd0dc4f3ee256bad5272948a64a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375616","number":375616,"commit":"f310f20702063b89e9573c37942c21d165620d46"}],"number":375722}
buildkite-pr-comment-->

dej611 committed:
Merge branch 'main' into fix/metric-schema

andrimal commented on src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/metric.ts:
```diff
@@ -121,13 +161,13 @@ const metricStatePrimaryMetricOptionsSchema = schema.object({
 
 const metricStateSecondaryMetricOptionsSchema = schema.object({
+  // this is used to differentiate primary and secondary metrics
+  // unfortunately givent he lack of tuple schema support we need to have some way
```
```suggestion
  // unfortunately given the lack of tuple schema support we need to have some way
```

andrimal reviewed (commented):
Just commenting on some typos for now. Will come back for a full review.

dej611 committed:
Apply suggestions from code review

Co-authored-by: Andreana Malama <andreana.malama@elastic.co>

dej611 committed:
Merge branch 'main' into fix/metric-schema

dej611 committed:
Merge branch 'main' into fix/metric-schema

dej611 committed:
Merge branch 'main' into fix/metric-schema

drewdaemon committed:
[Field formatters] make static-lookup formatter work with aggregated boolean fields (#249311)

## Summary

fix https://github.com/elastic/kibana/issues/100832

Explanation on parent issue:
https://github.com/elastic/kibana/issues/100832#issuecomment-3757365053


<img width="1663" height="918" alt="Screenshot 2026-01-15 at 4 41 34 PM"
src="https://github.com/user-attachments/assets/fea0c4c1-9c07-46fc-842b-2c3aae7be5f3"
/>


### Checklist

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

benakansara committed:
[Metric threshold rule] Add No data behavior options (#247669)

Resolves https://github.com/elastic/kibana/issues/247418

## Summary

This PR introduces "No Data Behavior" options for Metric Threshold
alerts, providing users with more granular control over how alerts
behave when data stops being reported. It also simplifies the UI by
consolidating two separate options related to no-data state and fixes a
limitation with count aggregation.

## Changes

### No Data Behavior Options

Replaced the previous two checkboxes ("Alert me if there's no data" and
"Alert me if a group stops reporting data") with a single, unified set
of three radio button options under "If there is no data":

1. **Recover if there's no data (default)** - Alert recovers when data
is missing
2. **Remain active if there's no data (new)** - Alert stays in its
current active state without transitioning to a no-data state
3. **Alert me if there's no data** - Triggers a NO_DATA alert when data
is missing

### UI Simplification

- **Consolidated two settings into one:** Previously, users had to
configure "Alert me if there's no data" for ungrouped alerts and "Alert
me if a group stops reporting data" for grouped alerts separately. Now,
a single "Alert me if there's no data" option handles both scenarios
automatically based on whether `groupBy` is enabled.
- **Added a dynamic tooltip** on the "Alert me if there's no data"
option that shows contextually appropriate help text:
- When `groupBy` is used: Explains the behavior for dynamically scaling
infrastructures
- When `groupBy` is not used: Explains the general no-data alert
behavior
 
### Bug Fix: Count Aggregation No Data Handling (similar to Custom
threshold rule)

- **Fixed:** Previously, alerts using count aggregation would not
properly detect "no data" conditions because a count of zero (0) was
treated as valid data rather than no data.

- **Change:** When count aggregation returns 0 and the threshold is not
met, the value is now returned as null (indicating no data) instead of
0. This aligns the behavior with other aggregation types and the Custom
Threshold rule.

### UI Changes

<img width="318" height="163" alt="Screenshot 2026-01-16 at 1 33 32 PM"
src="https://github.com/user-attachments/assets/7ea861f1-0d56-4b44-97f4-8f27aac06f31"
/>

---------

Co-authored-by: Panagiotis Moustafellos <pmoust@elastic.co>

jbudz committed:
[build/serverless] Add support for jemalloc (#247873)

Adds support for enabling jemalloc behind the `KBN_ENABLE_JEMALLOC` flag
on serverless projects.

---------

Co-authored-by: Tyler Smalley <tylersmalley@gmail.com>

lorenabalan committed:
[WorkplaceAI][Bugfix] Add datasource name in namespace (#249123)

## Summary

To unblock creating multiple sources of the same type. Not a huge fan of
this approach but will want to re-evaluate workflows interaction as part
of https://github.com/elastic/kibana/pull/248995

<img width="1228" height="289" alt="Screenshot 2026-01-15 at 10 48 23"
src="https://github.com/user-attachments/assets/97a49bf3-adc4-4919-ab48-eba5b774b698"
/>
<img width="1386" height="345" alt="Screenshot 2026-01-15 at 10 48 34"
src="https://github.com/user-attachments/assets/09d04b60-6357-42bd-8756-4396890a8de9"
/>
<img width="1201" height="388" alt="Screenshot 2026-01-15 at 10 49 06"
src="https://github.com/user-attachments/assets/29efd53a-fa2e-4f96-8a86-729ada715afa"
/>


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

andrimal committed:
[Lens as code] Fix the 'rankBy' in terms bucket operation transformation for charts with formula ref columns (#247449)

Fix #247436

This PR fixes the transformation issue of the `rankBy` field in a
`terms` bucket operation, by filtering out the formula reference columns
from the lens state, if present.

Check the PR satisfies following conditions.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

Supplementing committed:
[Fleet] Truncate package description in cards in browse experience  (#249092)

## Summary

Closes https://github.com/elastic/ingest-dev/issues/6754

Changes the maximum lines of the `description` to be 2 on package cards
in the browse experience so that longer descriptions are truncated,
which results in a more consistent layout, rather than some rows having
larger cards than others.

Before: 
<img width="1271" height="539" alt="image"
src="https://github.com/user-attachments/assets/041b584b-4083-4578-aeb1-3966b45c63bc"
/>
After: 

<img width="1173" height="483" alt="image"
src="https://github.com/user-attachments/assets/c0f81a74-bd5e-446b-a8fb-f20d13312886"
/>



### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

szwarckonrad committed:
[EDR Workflows] Fix trusted devices test error message expectation (#249413)

Fixes the trusted devices API integration test that was failing because
the error message expectation didn't match the actual validator output.

The test expected `String must contain at least 1 character` or `No
empty strings allowed` but the TrustedDeviceValidator returns `Field
value cannot be empty`.

This test was prematurely re-enabled in #249231 - the underlying issue
was never fixed when it was originally skipped.

Closes https://github.com/elastic/kibana/issues/249366
Closes https://github.com/elastic/kibana/issues/249365

qn895 committed:
[AI Infra] Add timeout parameter to InferenceChatModel (#248326)

## Summary

This PR addresses https://github.com/elastic/kibana/issues/221318 and
adds `timeout` parameter to InferenceChatModel as an optional parameter

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

lorenabalan committed:
[WorkplaceAI] Rename to DataSource and DataCatalog (#248837)

## Summary

Related to https://github.com/elastic/search-team/issues/12061
This renames 
1. all internal abstractions and references to be consistently named
DataSource(s) and DataCatalog
2. `data_connectors` plugin to `data_sources`
3. `data_sources_registry` plugin to `data_catalog`

**Also** removes `public` part of `data_catalog` plugin as we don't use
or need it and it's dead code.

TODO at a later date: 
⚠️ Need to look into what's needed to change the SO type name, as
migrations for those are more complicated.
⚠️ Rename a bunch of references in the new FE code.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

drewdaemon committed:
[ES|QL] fix filtering out null values (#249302)

## Summary

fix https://github.com/elastic/kibana/issues/246905



https://github.com/user-attachments/assets/308e3c5c-8241-4161-9252-114b343a61eb


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

### Release note

Fixes filtering out null values from the Discover histogram legend in
ES|QL mode.

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

kibanamachine committed:
skip failing test suite (#248578)

mistic committed:
skip flaky suite (#248030)

meghanmurphy1 committed:
[WorplaceAI] GitHub MCP Connector part of Data Source (#247659)

## Summary

The GitHub Data Source will now use the MCP connector rather than the
GitHub connector-v2 spec and bulk imports primitive tools that are
namespaced with the data source and create workflows for searching.

This PR: 
- separates the bulk import logic from the MCP connector API into an
exported function `bulkCreateMcpTools`
- adds `getNamedMcpTools` to get back the tools that are listed as part
of `importedTools` in the Github`DataTypeDefinition.`
- adds `createStackConnector `util to create the stack connector and
build the secrets based on the connector v2 spec or MCP config.
- update to use `credentials` in api instead of `token` for clarity that
it can be other forms of auth as well.
- added workflows that become workflow tools for searching that can
later have steps to sanitize/rerank.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jillguyonnet committed:
Enable enableAgentRollback feature flag (#249416)

## Summary

Make `enableAgentRollback` feature flag `true` by default.

Implementation PR: https://github.com/elastic/kibana/pull/247398
Backport to 9.3: https://github.com/elastic/kibana/pull/249250

### Identify risks

Agent rollback is mostly insulated from other existing flows: low
probability risk of affecting agent actions menus in Fleet UI.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
skip failing test suite (#249130)

mistic committed:
skip flaky suite (#247374)

mistic committed:
skip flaky suite (#249144)

mistic committed:
skip flaky suite (#249143)

sodhikirti07 committed:
[ML] Update packetbeat dns tunneling datafeed to include runtime mappings (#249317)

kibanamachine committed:
[api-docs] 2026-01-17 Daily api_docs build (#249470)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1320

klacabane committed:
[streams][significant events] features identification task (#249239)

## Summary

Handle the features identification as a background task instead of a
synchronous process.
The task is not triggered anywhere yet

```
// trigger
POST kbn:/internal/streams/<name>/features/_task
{
 "action": "schedule",
 "from": "<iso>",
 "to": "<iso>",
 "connector_id": "<connector>"
}

// check status
GET kbn:/internal/streams/<name>/features/_status
```

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

MadameSheema committed:
[Security Solution] Remove flaky `connectors.cy.ts` test with duplicate coverage (#249431)

## Summary

Removes the flaky Cypress test `connectors.cy.ts` and all related code
that was only used by this test.

### Background

This test has been **skipped since July 2024** (~18 months) due to
flakiness ([#187701](https://github.com/elastic/kibana/issues/187701)).

After investigation, the test functionality is already covered at more
appropriate testing layers, making this E2E test redundant.

### Why Delete Instead of Fix

The test validates:
1. Creating a Jira connector returns 200
2. Saving the connector configuration returns 200
3. The connector name appears in the dropdown
4. The field mapping displays correctly

**All of these are already tested at API and unit test levels** (see
duplicate coverage below), making this E2E test unnecessary overhead.

### Duplicate Coverage

| Cypress Test Assertion | Covered By | File |
|------------------------|------------|------|
| Jira connector creation returns 200 | **API Integration Test** |
`x-pack/platform/test/alerting_api_integration/security_and_spaces/group2/tests/actions/connector_types/jira.ts`
→ `'should return 200 when creating a jira action successfully'` |
| Cases configuration save returns 200 | **API Integration Test** |
`x-pack/platform/test/cases_api_integration/security_and_spaces/tests/trial/configure/post_configure.ts`
→ `'should create a configuration with mapping'` |
| Connector dropdown renders correctly | **Unit Test** |
`x-pack/platform/plugins/shared/cases/public/components/configure_cases/index.test.tsx`
→ `'it renders the Connectors'`, `'it renders with correct props'` |
| Field mapping displays correctly | **Unit Test** |
`x-pack/platform/plugins/shared/cases/public/components/configure_cases/mapping.test.tsx`
→ `'correctly maps fields'`, `'it shows mapping form group'` |

### Changes

**Deleted files:**
- `cypress/e2e/explore/cases/connectors.cy.ts` - the test file
- `cypress/tasks/configure_cases.ts` - tasks only used by this test
- `cypress/screens/configure_cases.ts` - selectors only used by the
above tasks

**Modified files:**
- `cypress/screens/all_cases.ts` - removed orphaned
`EDIT_EXTERNAL_CONNECTION` selector
- `cypress/tasks/all_cases.ts` - removed orphaned
`goToEditExternalConnection` function

kibanamachine committed:
[api-docs] 2026-01-18 Daily api_docs build (#249478)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1321

h88 committed:
regenerate expired localhost.p12 certificate for SSL tests (#248589)

Fixes: https://github.com/elastic/kibana/issues/248579

shahargl committed:
fix: dark mode on yaml editor (#249481)

## Summary
<img width="2556" height="1296" alt="CleanShot 2026-01-18 at 10 58 20"
src="https://github.com/user-attachments/assets/cb415ba7-ab31-414c-9ac9-3fbf219fa17c"
/>

Fixes the "Test step" modal to respect dark mode theme, making it
consistent with the "Test Workflow" modal.

Closes https://github.com/elastic/security-team/issues/15438

## Problem

When using the inline "Run step" execution on a step that references
workflow input data, the "Test step" modal loses dark mode styling. The
modal's JSON editor turns white, breaking the dark theme.

The "Test Workflow" modal (full workflow run) correctly maintains dark
mode - the bug only affected the "Test step" modal for inline step
execution.

| Modal | Before | After |
|-------|--------|-------|
| **Test Workflow** | Dark mode works ✅ | Dark mode works ✅ |
| **Test step** | White background ❌ | Dark mode works ✅ |

## Root Cause

The `TestStepModal` component was using a hardcoded theme
(`'workflows-subdued'`) for the Monaco code editor, which doesn't
respond to dark mode changes.

The `WorkflowExecuteModal` (Test Workflow) correctly uses
`WORKFLOWS_MONACO_EDITOR_THEME` with the `useWorkflowsMonacoTheme()`
hook, which dynamically registers theme colors based on the current EUI
color mode.

## Solution

Updated `TestStepModal` to use the same theme system as
`WorkflowExecuteModal`:

1. Import and call `useWorkflowsMonacoTheme()` hook to register the
dynamic theme
2. Use `WORKFLOWS_MONACO_EDITOR_THEME` constant instead of hardcoded
`'workflows-subdued'`

## Testing

1. Enable dark mode in Kibana
2. Create a workflow with inputs defined
3. Add a step that references the input (e.g., `{{ inputs.message }}`)
4. Click the inline "Run step" button on that step
5. Verify the "Test step" modal maintains dark mode styling

talboren committed:
[One Workflow] Add connector icons for AbuseIPDB, GreyNoise, Shodan, and URLVoid (#249306)

## Summary

Adds missing connector icons for four threat intelligence connectors to
improve their visibility in the Workflows UI.

## Changes

- **AbuseIPDB**: Added SVG logo icon
- **GreyNoise**: Added PNG icon
- **Shodan**: Added PNG logo icon  
- **URLVoid**: Added PNG icon
- Updated `connector_icons_map.ts` to register all four new icons

## Visual Impact

These connectors will now display their respective brand icons in the
Workflows connector selection UI instead of generic placeholders.

shahargl committed:
fix: modal input (#249480)

## Summary

<img width="1880" height="934" alt="CleanShot 2026-01-18 at 10 42 43"
src="https://github.com/user-attachments/assets/73121ae0-ccaf-4d22-afe3-8394cbf8df40"
/>


Fixes the "Test step" modal to pre-populate input fields with
workflow-defined default values instead of showing generic placeholder
text.

Closes https://github.com/elastic/security-team/issues/15439

## Problem

When using the inline "Run step" execution, the test input modal showed
generic placeholder text (`"replace with your data"`) for workflow
inputs, even when those inputs had defined default values. The "Test
Workflow" modal correctly used default values, creating an inconsistent
experience.

**Example workflow:**
```yaml
inputs:
  - name: message
    default: "hello world"
    type: string

steps:
  - name: log_input
    type: console
    with:
      message: "{{ inputs.message }}"
```

| Modal | Before | After |
|-------|--------|-------|
| **Test Workflow** | `"hello world"` ✅ | `"hello world"` ✅ |
| **Test step** | `"replace with your data"` ❌ | `"hello world"` ✅ |

## Solution

The `buildContextOverride` function was only receiving `consts` and
`workflow` as static data, but not the workflow's `inputs` definitions.
When resolving `{{ inputs.* }}` references, it couldn't find default
values and fell back to generic placeholders.

**Changes:**
- Extended `buildContextOverride` to accept an optional
`inputsDefinition` parameter containing workflow input definitions with
their defaults
- Updated `useContextOverrideData` hook to pass the workflow's inputs
definition
- Added helper function `buildInputsFromDefinition` to convert input
definitions into a lookup object

**Behavior:**
- If an input has a `default` value → pre-populate with that value
- If an input has no `default` → fall back to `"replace with your data"`

## Testing

Added unit tests covering:
- Using input default values when `inputsDefinition` is provided
- Falling back to placeholder for inputs without defaults  
- Handling array input defaults
- Handling choice input defaults

kibanamachine committed:
[ES|QL] Update grammars (#249493)

This PR updates the ES|QL grammars (lexer and parser) and PromQL
grammars to match the latest version in Elasticsearch.

kibanamachine committed:
[api-docs] 2026-01-19 Daily api_docs build (#249496)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1322

jeramysoucy committed:
Add Elastic map-related URLs to default report-only CSP (#247131)

## Summary

This PR augments the default report-only connect-src and img-src to
allow Elastic map-related sources.


### Testing

1. Start ES & KB
2. Add web log sample data
3. Open the browser's dev tools console
4. Go to the web logs dashboard
5. Verify there are no "report-only" connect-src and img-src violations
for map-related Elastic.co URLs
6. Repeat the test from main and verify that there are "report-only"
connect-src and img-src violations for map-related Elastic.co URLs

shahargl committed:
fix: one year ago workflow execution (#249487)

before:
<img width="730" height="580" alt="CleanShot 2026-01-18 at 16 53 49"
src="https://github.com/user-attachments/assets/5542118b-2ee6-4de1-bb8b-8c4203fb3979"
/>

after:
<img width="725" height="584" alt="CleanShot 2026-01-18 at 17 00 21"
src="https://github.com/user-attachments/assets/64790b0d-6281-41b6-b714-c6772d73b6aa"
/>

## Summary

Fixes misleading "last year" date display in workflow execution list
when dates cross year boundaries.

**Issue:** https://github.com/elastic/security-team/issues/15310

### Problem

The "Last run" column in the workflows list was displaying "last year"
for executions that happened only 2-3 weeks ago, simply because they
crossed the calendar year boundary (e.g., Dec 20, 2025 viewed on Jan 10,
2026).

### Solution

Added a fix to `FormattedRelativeEnhanced`: when `selectUnit` from
`@formatjs/intl-utils` returns "year" but less than 180 days (6 months)
have actually passed, we use a more appropriate unit (weeks or months)
instead.

### Changes

**`formatted_relative_enhanced.tsx`:**
- Added year boundary fix: converts "year" to "weeks" or "months" when <
180 days have passed
- Refactored tooltip logic into separate `WithTooltip` component to
avoid calling `useFormattedDateTime` hook when `fullDateTooltip=false`
(fixes unnecessary hook call and improves testability)

**`status_badge.tsx`** and **`workflow_execution_list_item.tsx`:**
- Use `FormattedRelativeEnhanced` instead of `FormattedRelative`

**`status_badge.test.tsx`:**
- Added tests for year boundary scenarios

### Before

| Execution Date | Current Date | Display |
|----------------|--------------|---------|
| Dec 20, 2025 | Jan 10, 2026 | "last year" |
| Nov 10, 2025 | Jan 10, 2026 | "last year" |

### After

| Execution Date | Current Date | Display |
|----------------|--------------|---------|
| Dec 20, 2025 | Jan 10, 2026 | "3 weeks ago" |
| Nov 10, 2025 | Jan 10, 2026 | "2 months ago" |

All other relative time displays (weeks, days, hours, etc.) remain
unchanged.

## Test Plan

- [x] Unit tests verify dates crossing year boundaries don't show "last
year" when < 180 days old
- [ ] Manual verification: View workflow execution list with executions
from late December when current date is early January
- [ ] Verify "last week", "2 weeks ago" etc. still work as before

rosomri committed:
[One Workflow]: bugfix - Inputs section incorrectly selectable as "test step" (#249486)

eokoneyo committed:
fix issue with eui flyouts not rendering appropriately in new layout (#248921)

## Summary

Fixes https://github.com/elastic/kibana/issues/245851 for export flyouts

<img width="717" height="883" alt="Screenshot 2026-01-13 at 22 38 35"
src="https://github.com/user-attachments/assets/a9056904-3f78-4cae-8ce2-9ec8fd526a9b"
/>

Also accounts for when the below header option is selected in eui
flyouts;

<img width="499" height="832" alt="Screenshot 2026-01-13 at 22 39 46"
src="https://github.com/user-attachments/assets/0585998f-dd9c-447a-a7ff-12d32deda9d9"
/>


<!--
### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

-->

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

kibanamachine committed:
[One Discover][Logs UX] Update OpenTelemetry Semantic Conventions (#249498)

This PR updates the OpenTelemetry semantic conventions definitions to
the latest version.

### Changes Made
- ✅ Updated `resolved-semconv.yaml` with latest conventions from
OpenTelemetry repository
- ✅ Regenerated `resolved-semconv.ts` with new field definitions and
metadata
- ✅ Applied ESLint formatting to generated files

### Validation
- ✅ TypeScript compilation successful
- ✅ Package tests passing
- ✅ ESLint formatting applied

🤖 Generated automatically by Buildkite workflow

skynetigor committed:
[One Workflow] Simplify the default workflow template (#249072)

## Summary

Simplify the default workflow template for new workflows by removing
advanced example features and providing a minimal "Hello World" starting
point. This change makes the workflow creation experience more
approachable for new users by presenting them with a basic,
easy-to-understand template instead of a comprehensive example with
loops, conditionals, inputs, constants, and filters.

## New Features

- Simplify workflow template: Provide a minimal default workflow
template with just a single console log step

## Usage Example

When creating a new workflow, users will now see this minimal template:

```yaml
name: New workflow
enabled: false
description: This is a new workflow
triggers:
  - type: manual

inputs:
  - name: message
    type: string
    default: "hello world"

steps:
  - name: hello_world_step
    type: console
    with:
      message: "{{ inputs.message }}"
```

maryam-saeidi committed:
[Lens] Display row numbers for the data table by default (#247834)

Closes https://github.com/elastic/kibana/issues/158832

## Summary

This PR adds a new style configuration to add row numbering for the
table:

<img width="1508" height="863" alt="image"
src="https://github.com/user-attachments/assets/b1c2a6ea-6882-4d9c-8488-10d998bd5571"
/>


By default, it is enabled for any new table that is added but disabled
for the previously defined tables, as shown below:


https://github.com/user-attachments/assets/aaf45ed1-81e4-47bd-af7f-8a98cc94b1b8

I also checked the Cases flow and it worked as expected, too:


https://github.com/user-attachments/assets/29a9fe92-da21-48b8-9e6e-0b686ccd0ceb
